app/Http/Controllers/Api/V1/MicroSoftTeamController.php

```diff
@@ -0,0 +1,222 @@
+            
+            $isUpdated = $this->userRepository->update([
+                'msteam_access_token' => $accessToken
+            ], $request->get('state'));
+            
+            if ($isUpdated) {
+                return response([
+                    'message' => 'Access token has been saved successfully.',
+                ], 200);
+            }
+            return response([
+                'errors' => ['Failed to save the token.'],
+            ], 500);
+
+        } else {
+
+            $gid = $request->get('gid');
+            $url = $this->microsoftTeamRepository->getLoginUrl($gid);
+            
+            return Redirect::to($url);
+        }
+   
+    }
+    
+    /**
+	 * Get List of Classes
+	 *
+	 * Get List of Microsoft Team Classes
+	 
+     * @response  200 {
+     *   "classes": Array of classes
+     * }
+     
+     *
+     * @param MSTeamCreateClassRequest $createClassRequest
+     * @return Response
+	 */
+    public function getClasses(Request $request)
+    {
+        $authUser = auth()->user();
+        $token = $authUser->msteam_access_token;
+        
+        return response([
+            'classes' => $this->microsoftTeamRepository->getClassesList($token),
+        ], 200);
+    }
+
+    /**
+	 * Create a new Class
+	 *
+	 * Create a new Class/Team into Microsoft Team
+	 *
+     * @urlParam project required The Id of a project. Example: 9
+     * @bodyParam displayName required string Name of the class. Example: Test Class
+     * @bodyParam access_token string|null The stringified of the GAPI access token JSON object
+     
+     * @response  200 {
+     *   "message": [
+     *     "Class have been created successfully"
+     *   ]
+     * }
+     
+     *
+     * @param MSTeamCreateClassRequest $createClassRequest
+     * @return Response
+	 */
+    public function createMsTeamClass(MSTeamCreateClassRequest $createClassRequest)
+    {
+        
+        $data = $createClassRequest->validated();
+        $authUser = auth()->user();
+        $token = $authUser->msteam_access_token;
+        $statusCode = $this->microsoftTeamRepository->createMsTeamClass($token, $data);
+
+        if($statusCode === 202) {
+            return response([
+                'message' => 'Class have been created successfully',
+            ], 200);
+        }
+        
+        return response([
+            'errors' => 'Something wrong happened.',
+        ], 500);
+        
+    }
+
+    /**
+	 * Publish project
+	 *
+	 * Publish the project activities as an assignment
+	 *
+     * @urlParam MSTeamCreateAssignmentRequest $createAssignmentRequest required The Id of a project. Example: 9
+     * @bodyParam classId required string Id of the class. Example: Test Class
+     
+     * @response  200 {
+     *   "message": [
+     *     "Class have been created successfully"
+     *   ]
+     * }
+     
+     *
+     * @param MSTeamCreateClassRequest $createClassRequest
```

Comment:
In signature "MSTeamCreateAssignmentRequest" is passed instead of "MSTeamCreateClassRequest"